### DISCUSSION TITLE
setting default values from axios get request

### DISCUSSION THREAD

**Discussion by eranamarante** [react-hook-form/react-hook-form#2869] [Q&A] [answered]
Hello, new to react-hook-form. Just need some help in setting default values.

Here's a [sandbox](https://codesandbox.io/s/gifted-leaf-y5b4k?file=/src/App.js)

I have 2 example of what I tried.
1. defaultValues = response data from axios (not working or I should not do this?)
2. setValue of the specific input 

what should I use? although setValue is working. Im just wondering if i can set a value of whole object for ex:

setValue(someObjectFromAxios);

----

sample response:
```
{
 customer_id: 1,
 name: 'John Doe',
 company: 'Company ABC',
 address: 'Somewhere in neverland'
 ...
}
```

Thanks

**Comment by bluebill1049** [comment, marked as answer]
you can use reset API to set your form values when your data is resolved.

**Reply by bluebill1049** [reply]
https://react-hook-form.com/api#useForm 
read the defaultValues section

**Reply by eranamarante** [reply]
OMG thank you so much!, I didn't know I can use reset as default values
